utils/debug/Debug.js

```diff
@@ -95,6 +108,81 @@ function Debug(view, datDebugTool, chartDivContainer) {
+    const debugCamera = view.camera.camera3D.clone();
+    debugCamera.fov *= 1.5;
+    debugCamera.updateProjectionMatrix();
+    const g = view.mainLoop.gfxEngine;
+    const r = g.renderer;
+    let fogDistance = view.fogDistance;
+    helper.visible = false;
+    view.scene.add(helper);
+
+    function tileDisplay(obj) {
+        if (obj.setFog && fogDistance) {
+            obj.setFog(fogDistance);
+        }
+    }
+
+    const bClearColor = new Color();
+    function renderCameraDebug() {
+        if (state.debugCameraWindow && debugCamera) {
+            const size = { x: g.width * 0.2, y: g.height * 0.2 };
+            debugCamera.aspect = size.x / size.y;
+            const camera = view.camera.camera3D;
+            const coord = new Coordinates(view.referenceCrs, camera.position).as(tileLayer.extent._crs);
+            const altitude = 1.5 * coord._values[2];
+            if (altitude > 1) {
+                coord._values[2] = altitude;
+                const position = coord.as(view.referenceCrs).xyz();
+                camera.worldToLocal(position);
+                position.z += altitude;
+                camera.localToWorld(position);
+                debugCamera.position.copy(position);
+                const lookAt = view.camera.camera3D.position.clone();
+                camera.worldToLocal(lookAt);
+                lookAt.z -= altitude * 1.5;
+                camera.localToWorld(lookAt);
+                debugCamera.lookAt(lookAt);
+            } else {
+                debugCamera.position.set(0, 0, 100);
+                camera.localToWorld(debugCamera.position);
+                debugCamera.lookAt(camera.position);
+            }
+
+            debugCamera.updateProjectionMatrix();
+            if (view.atmosphere) {
+                view.atmosphere.visible = false;
+            }
+            fogDistance = 10e10;
+            for (const obj of tileLayer.level0Nodes) {
+                obj.traverseVisible(tileDisplay);
+            }
+            helper.visible = true;
+            helper.updateMatrixWorld(true);
+            bClearColor.copy(r.getClearColor());
+            r.setViewport(g.width - size.x, g.height - size.y, size.x, size.y);
+            r.setScissor(g.width - size.x, g.height - size.y, size.x, size.y);
+            r.setScissorTest(true);
+            r.setClearColor(0xeeeeee);
+            r.clear();
+            r.clearDepth();
+            r.render(view.scene, debugCamera);
+            r.setScissorTest(false);
+            r.setClearColor(bClearColor);
+            helper.visible = false;
+            if (view.atmosphere) {
+                view.atmosphere.visible = true;
+            }
+            fogDistance = view.fogDistance;
+            for (const obj of tileLayer.level0Nodes) {
+                obj.traverseVisible(tileDisplay);
+            }
+        }
+    }
+    view.addFrameRequester(MAIN_LOOP_EVENTS.AFTER_RENDER, renderCameraDebug);
```

Comment:
Oops sorry, just one more change: could you dynamically add/remove this FrameRequester instead of always adding it? (just need to be done in the `debugCameraWindow` `onChange` function. Thanks

Reply:
Done